IssueTracker/DataAccess/DbAccess/SQLDataAccess.cs

```diff
@@ -14,24 +18,501 @@ public SQLDataAccess(IConfiguration config)
             _config = config;
         }
 
-        public async Task<IEnumerable<T>> LoadDataAsync<T>(
+        public async Task<IEnumerable<TFirst>> LoadDataAsync<TFirst>(
             string storedProcedure,
             string connectionId = "Default")
         {
             using (IDbConnection connection = new SqlConnection(_config.GetConnectionString(connectionId)))
             {
-                return await connection.QueryAsync<T>(storedProcedure,
+                return await connection.QueryAsync<TFirst>(storedProcedure,
                     commandType: CommandType.StoredProcedure);
             }
         }
-        public async Task<IEnumerable<T>> LoadDataAsync<T, U>(
+        public async Task<IEnumerable<TFirst>> LoadDataAsync<TFirst, TParameter>(
             string storedProcedure,
-            U parameters,
+            TParameter parameter,
             string connectionId = "Default")
         {
             using (IDbConnection connection = new SqlConnection(_config.GetConnectionString(connectionId)))
             {
-                return await connection.QueryAsync<T>(storedProcedure, parameters,
+                return await connection.QueryAsync<TFirst>(storedProcedure, parameter,
+                    commandType: CommandType.StoredProcedure);
+            }
+        } //TODO: de gandit putin la aceste metode!
+        public async Task<IEnumerable<TFirst>> LoadDataAsync<TFirst, TSecond>(string storedProcedure, string connectionId = "Default")
```

Comment:
can't everything be resolved with just one method? i.e. use a list of types and resolve with reflection?

Reply:
I will try to solve this.